src/browser/mime.zig

```diff
@@ -52,6 +54,100 @@ pub const Mime = struct {
         other: struct { type: []const u8, sub_type: []const u8 },
     };
 
+    /// Returns the null-terminated charset value.
+    pub inline fn charsetString(mime: *const Mime) [:0]const u8 {
+        return @ptrCast(&mime.charset);
+    }
+
+    /// Removes quotes of value if quotes are given.
+    ///
+    /// Currently we don't validate the charset.
+    /// See section 2.3 Naming Requirements:
+    /// https://datatracker.ietf.org/doc/rfc2978/
+    fn parseCharset(value: []const u8) error{ CharsetTooBig, Invalid }![]const u8 {
+        // Cannot be larger than 40.
+        // https://datatracker.ietf.org/doc/rfc2978/
+        if (value.len > 40) return error.CharsetTooBig;
+
+        // If the first char is not a quote, value can be used directly.
+        // Whitespace is not allowed.
+        if (value[0] != '"') {
+            return value;
+        }
+
+        // Search for second quote begins.
+        // Skip the first character.
+        var offset: usize = 1;
+
+        // Charset values are not so large; 128-bit registers should be
+        // more than enough.
+        const vec_size = 16;
+        const Vec = @Vector(vec_size, u8);
+        const UInt = std.meta.Int(.unsigned, vec_size);
+        const block_size = @sizeOf(u64);
+
+        const charset = blk: {
```

Comment:
If the value is quoted, is the point to escape special characters, and thus this should un-escape them? Which I don't think it is.

I think this is just finding the (non-matching) closing quote, and stripping it off. If that's all we want, surely the faster and simplest thing to do is just look at the last character of the value?

Reply:
`try parseCharset("\"ut\\\"f-8\"")`  as a test, should return `ut"f-8`   but this returns `ut\`

Reply:
It's fine if we don't escape it [until we have a use case that needs it]. But I have to imagine a simple reverse loop will be faster for the _very_ common case that `"` will be the last character.